pkg/terraform/exec/plugins/vsphereprivate/resource_vsphereprivate_import_ova.go

```diff
@@ -0,0 +1,384 @@
+			return nil, err
+		}
+		if network == networkObjectName {
+			importOvaParams.Network = networkObj
+			break
+		}
+	}
+
+	// Find all the datastores that are configured under the cluster
+	datastores, err := clusterComputeResource.Datastores(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Find the specific datastore by the name provided
+	for _, datastoreObj := range datastores {
+		datastoreObjName, err := datastoreObj.ObjectName(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if datastore == datastoreObjName {
+			importOvaParams.Datastore = datastoreObj
+			break
+		}
+	}
+
+	// Find all the HostSystem(s) under cluster
+	hosts, err := clusterComputeResource.Hosts(ctx)
+	if err != nil {
+		return nil, err
+	}
+	foundDatastore := false
+	foundNetwork := false
+	var hostSystemManagedObject mo.HostSystem
+
+	// Confirm that the network and datastore that was provided is
+	// available for use on the HostSystem we will import the
+	// OVA to.
+	for _, hostObj := range hosts {
+		hostObj.Properties(ctx, hostObj.Reference(), []string{"network", "datastore"}, &hostSystemManagedObject)
+
+		if err != nil {
+			return nil, err
+		}
+		for _, dsMoRef := range hostSystemManagedObject.Datastore {
+
+			if importOvaParams.Datastore.Reference().Value == dsMoRef.Value {
+				foundDatastore = true
+				break
+			}
+		}
+		for _, nMoRef := range hostSystemManagedObject.Network {
+			if importOvaParams.Network.Reference().Value == nMoRef.Value {
+				foundNetwork = true
+				break
+			}
+		}
+
+		if foundDatastore && foundNetwork {
+			importOvaParams.Host = hostObj
+			resourcePool, err := hostObj.ResourcePool(ctx)
+			if err != nil {
+				return nil, err
+			}
+			importOvaParams.ResourcePool = resourcePool
+		}
+	}
+	if !foundDatastore {
+		return nil, errors.Errorf("failed to find a host in the cluster that contains the provided datastore")
+	}
+	if !foundNetwork {
+		return nil, errors.Errorf("failed to find a host in the cluster that contains the provided network")
+	}
+
+	return importOvaParams, nil
+}
+
+// Used govc/importx/ovf.go as an example to implement
+// resourceVspherePrivateImportOvaCreate and upload functions
+// See: https://github.com/vmware/govmomi/blob/cc10a0758d5b4d4873388bcea417251d1ad03e42/govc/importx/ovf.go#L196-L324
+func upload(ctx context.Context, archive *importx.ArchiveFlag, lease *nfc.Lease, item nfc.FileItem) error {
+	file := item.Path
+
+	f, size, err := archive.Open(file)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	opts := soap.Upload{
+		ContentLength: size,
+	}
+
+	return lease.Upload(ctx, item, f, opts)
+}
+
+func resourceVSpherePrivateImportOvaCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning import ova create", d.Get("filename").(string))
+
+	// Since this is the same context that will be used for upload
+	// it should only set to TODO since we do not know how long
+	// it may take to upload the RHCOS ova to vSphere.
+	ctx := context.TODO()
```

Comment:
shouldn't we have all requests use 5 min timeout context and the upload one use a separate TODO/Background context..